### PR TITLE
Integrate pull request preview environments 

### PR DESCRIPTION
**Pull request by waveywaves**
This PR adds a github workflow which will build the cockpit image and push it to the registry.uffizzi.com registry which is an ephemeral images registry. This image will be then used by Uffizzi to deploy the preview for the PR. Initially we are hoping this works as a bastion cockpit instance for you but in subsequent PRs we can work on standing up another instance of cockpit which will work as a normal server with cockpit installed. 

This contents of this PR have been tested against my own fork of cockpit https://github.com/waveywaves/cockpit/pull/1. 
After this PR is merged, you can expect all subsequent PRs to this repo to have a a comment like https://github.com/waveywaves/cockpit/pull/1#issuecomment-1374946004 pinned which will contain the link to the preview environment url like https://app.uffizzi.com//github.com/waveywaves/cockpit/pull/1 which was created by the github action workflow.

If you want more information on the github action workflow and why it has been written in two different files the way it is, I would suggest you to checkout a blogpost we wrote as an explainer over [here](https://www.uffizzi.com/preview-environments-guide/github-actions-two-stage-workflow).

related issue https://github.com/cockpit-project/cockpit/issues/18047

### CODE REVIEW

**Review by KKoukiou — COMMENTED** [review]
Hi :) First thanks for taking the time to implement this. 

I would say that making this default run with each PR is a bit overkill and unnecessary waste of uffizzis resources. I would like to talk about it with the rest of the cockpit team, but I feel on demand run of uffizzi workflow with listening for some comment is the direction I would prefer.

@martinpitt any feedback  on this?

File: .github/workflows/uffizzi-build.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
+name: Build PR Image
+on:
+  pull_request:
+    types: [opened,synchronize,reopened,closed]
```
Comment (martinpitt) [inline .github/workflows/uffizzi-build.yaml:4]: I agree with @KKoukiou about not triggering this for every PR run. And this is especially not needed on "closed".

We used to trigger this with a magic comment `/livetest`, see the [old workflow](https://github.com/cockpit-project/cockpit/blob/266/.github/workflows/livetest.yml). Something similar could be done here.

Comment (martinpitt) [reply .github/workflows/uffizzi-build.yaml:4]: Ah, nevermind, "closed" is used for cleaning up. But what about cleaning up old instances from force pushes? Don't they pile up?

File: .github/workflows/uffizzi-build.yaml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    name: Build and Push `cockpit`
+    runs-on: ubuntu-latest
+    if: ${{ github.event_name != 'pull_request' || github.event.action != 'closed' }}
+    outputs:
+      tags: ${{ steps.meta.outputs.tags }}
+    steps:
+      - name: Checkout git repo
+        uses: actions/checkout@v3
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
```
Comment (martinpitt) [inline .github/workflows/uffizzi-build.yaml:18]: I had a look at https://github.com/docker/setup-buildx-action , but this seems very generic. It wraps https://github.com/docker/buildx, which again wraps https://github.com/moby/buildkit, i.e. this very quickly gets to the point when this is completely non-understandable nor reproducible locally, and presumably will also take a lot longer than necessary. We already know how to build cockpit, and should re-use that here.

I'm afraid I don't yet know anything about the uffizi environment (not even whether it's deb, rpm, arch, or something else), so that will take some time to learn.

Did you test this on a fork? How do these environments look like?

Thanks!

Comment (martinpitt) [reply .github/workflows/uffizzi-build.yaml:18]: How do you log in there? I.e. which user/password/host? I don't see this on https://github.com/UffizziCloud/uffizzi. So far it looks like it's running a cockpit/ws container in bastion mode, but that's usually not the interesting part. Thanks!

File: .github/workflows/uffizzi-build.yaml (hunk 1 above)
Comment (martinpitt) [reply .github/workflows/uffizzi-build.yaml:4]: Sure, you can/should use the default GITHUB_TOKEN for that. However, I'm happy to work on this part, I've done it before and it's not hard. So it's fine for you to leave the trigger alone in this PR.

File: .github/workflows/uffizzi-build.yaml (hunk 2 above)
Comment (martinpitt) [reply .github/workflows/uffizzi-build.yaml:18]: @waveywaves : I'm afraid I don't understand your answer. Right now, on https://app.uffizzi.com//github.com/waveywaves/cockpit/pull/1 I see the cockpit login page. In order to to see the actual cockpit session, one  needs to log into the machine which has the packages/pages from the PR installed. Otherwise, the only thing you could test was the login page itself (which doesn't change often, and isn't very interesting). So what user/password do I need to type there to log into the temporary server?

**Comment by martinpitt** [comment]
This is quite stale, and has some unresolved important questions open. Do you still care about this?
